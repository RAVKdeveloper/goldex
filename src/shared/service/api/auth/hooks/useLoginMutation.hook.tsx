import { useMutation } from '@tanstack/react-query'

import { authApi } from '../api/auth.api'

export const useLoginMutation = () => {
  return useMutation({
    mutationFn: authApi.login,
    mutationKey: ['login user'],
  })
}
