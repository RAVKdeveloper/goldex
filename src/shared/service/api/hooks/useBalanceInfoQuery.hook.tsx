import { useQuery } from '@tanstack/react-query'

import { BalancesService } from '../../client'

export const useBalanceInfoQuery = () => {
  return useQuery({
    queryKey: ['getBalances'],
    queryFn: BalancesService.getOwnBalance,
  })
}
