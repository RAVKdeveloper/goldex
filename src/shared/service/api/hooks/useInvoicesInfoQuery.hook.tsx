import { useQuery } from '@tanstack/react-query'
import { InvoicesService } from '../../client'

interface useInvoicesInfoQueryType {
  page: number
  size: number
}

export const useInvoicesInfoQuery = ({ page = 1, size = 8 }: useInvoicesInfoQueryType) => {
  return useQuery({
    queryKey: ['invoicesInfo', page],
    queryFn: () => InvoicesService.readInvoicesList({ page, size }),
  })
}
