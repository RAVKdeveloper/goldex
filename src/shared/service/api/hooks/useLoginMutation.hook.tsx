import { useMutation } from '@tanstack/react-query'
import { LoginService } from '../../client'

export const useLoginMutation = () => {
  return useMutation({
    mutationFn: LoginService.login,
    mutationKey: ['loginUser'],
  })
}
