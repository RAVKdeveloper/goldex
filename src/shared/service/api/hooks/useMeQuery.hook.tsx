import { useQuery } from '@tanstack/react-query'
import { UsersService } from '../../client'

export const useMeQuery = () => {
  return useQuery({
    queryKey: ['getAuthMe'],
    queryFn: UsersService.getMyData,
  })
}
