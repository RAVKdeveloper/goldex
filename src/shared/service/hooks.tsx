export { useLoginMutation } from './api/hooks/useLoginMutation.hook'
export { useMeQuery } from './api/hooks/useMeQuery.hook'
export { useBalanceInfoQuery } from './api/hooks/useBalanceInfoQuery.hook'
