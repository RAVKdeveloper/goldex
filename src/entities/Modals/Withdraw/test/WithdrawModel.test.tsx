import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import * as reduxHooks from '../../../../shared/service/store/hooks/hooks'
import { WithdrawModalEntity } from '../ui/WithdrawModal.entity'

jest.mock('react-redux')

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

describe('testing WithdrawModalEntity component', () => {
  it('render and typing', () => {
    jest.spyOn(reduxHooks, 'useAppDispatch')
    jest.spyOn(reduxHooks, 'useAppSelector').mockReturnValue({ isOpen: true })

    const user = userEvent.setup()

    const component = render(<WithdrawModalEntity />)

    const modalComponent = screen.getByTestId('modal')
    const amountInput = screen.getByTestId('amount')
    const addressInput = screen.getByTestId('address')

    user.type(amountInput, '3243432')
    user.type(addressInput, 'Tlwqwrhi35fkfk93j')

    expect(component).toMatchSnapshot()
    expect(modalComponent).toBeInTheDocument()
  })
})
