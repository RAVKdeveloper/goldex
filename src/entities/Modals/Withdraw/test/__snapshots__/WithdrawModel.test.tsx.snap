// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing WithdrawModalEntity component render and typing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper none"
      >
        <modal
          data-testid="modal"
        >
          <div
            class="popup"
          >
            <modalheader
              paragraph=""
              title="modal.withdraw"
            />
            <form
              class="form"
            >
              <textfield
                data-testid="amount"
                label="labels.amount"
                name="amount"
              />
              <textfield
                data-testid="address"
                label="labels.address"
                name="address"
              />
              <modalactions
                btntextaction="btn.withdraw"
                btntextcancel="btn.cansel"
              />
            </form>
          </div>
        </modal>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper none"
    >
      <modal
        data-testid="modal"
      >
        <div
          class="popup"
        >
          <modalheader
            paragraph=""
            title="modal.withdraw"
          />
          <form
            class="form"
          >
            <textfield
              data-testid="amount"
              label="labels.amount"
              name="amount"
            />
            <textfield
              data-testid="address"
              label="labels.address"
              name="address"
            />
            <modalactions
              btntextaction="btn.withdraw"
              btntextcancel="btn.cansel"
            />
          </form>
        </div>
      </modal>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
